src/Database.hs

```diff
@@ -145,15 +145,15 @@ updateLastBlock ::
   ) =>
   BlockInfo ->
   m ()
-updateLastBlock (BlockInfo slot hash) = do
+updateLastBlock blockInfo = do
   let sql = "UPDATE last_block SET slot = $1, hash = $2"
       enc =
         mconcat
           [ fst >$< Encoders.param (Encoders.nonNullable Encoders.int8)
           , snd >$< Encoders.param (Encoders.nonNullable Encoders.text)
           ]
       dec = Decoders.noResult
-      stmt = Session.statement (slot, hash) $ Statement sql enc dec True
+      stmt = Session.statement (blockSlot blockInfo, blockId blockInfo) $ Statement sql enc dec True
   dbConnection <- ask
   res <- liftIO $ Session.run stmt dbConnection
   case res of
```

Comment:
GH doesn't allow to comment in right place. `getLastBlock` can be changed to return slot and hash instead of `BlockInfo`.